lms/services/lti_h.py

```diff
@@ -63,7 +53,7 @@ def _sync_to_h(self, groups):
             self._h_api.upsert_group(group)
 
             self._group_info_service.upsert(
-                authority_provided_id=self._context.h_group.authority_provided_id,
+                authority_provided_id=group.authority_provided_id,
```

Comment:
Yikes. This looks like it was wrong before: sending the same `authority_provided_id` for every group! But it worked because of course there is only one group and it's always `self._context.h_group`. It wouldn't work with section groups now. Anyway, this fixes the latent bug